eeps/eep-0061.md

```diff
@@ -0,0 +1,255 @@
+
+Example 2. Python, mypy:
+
+    def foo(n: object) -> int:
+        return n
+               ~ Incompatible return value type (got "object", expected "int")
+
+    def n_fun(n: int) -> int:
+        return n
+
+    def foo_n(n: object) -> int:
+        return n_fun(n)
+                     ~ Argument 1 to "n_fun" has incompatible type "object";
+                                                  expected "int"  [arg-type]
+
+Example 3. TypeScript:
+
+    function foo(x: unknown): number {
+        return x;
+        ~~~~~~~~~ Type 'unknown' is not assignable to type 'number'.(2322)
+    }
+
+    function n_fun(n: number): number {
+        return n;
+    }
+
+    function foo_n(n: unknown): number {
+        return n_fun(n);
+                     ~ Argument of type 'unknown' is not assignable
+                                      to parameter of type 'number'
+    }
+
+`dynamic()` type
+========
+
+eqWAlizer has a special type `eqwalizer:dynamic()` which is documented
+[here](https://github.com/WhatsApp/eqwalizer/blob/main/docs/reference/modes.md#type-eqwalizerdynamic).
+
+The type is similar to the following types in other languages and type-checkers:
+
+- [Any](https://docs.python.org/3/library/typing.html#the-any-type) in Python (mypy, PyRight, pyre handle it the same way)
+- [any](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#any) in TypeScript
+- [untyped](https://github.com/ruby/rbs/blob/master/docs/syntax.md) in Ruby/rbs
+- [untyped](https://sorbet.org/docs/untyped) in Ruby/Sorbet
+- [any](https://luau-lang.org/typecheck#any-type) in luau
+- [dynamic](https://docs.hhvm.com/hack/built-in-types/dynamic) in Hack
+- [any](https://flow.org/en/docs/types/any/) in Flow
+
+eqWAlizer borrowed the name from Hack.
+The naming choice was dictated by the following considerations:
+
+  - `eqwalizer:any()` would misleading and confusion because of the builtin Erlang type `any()`
+  - `eqwalizer:dynamic()` is often used to mark "inherently dynamic code":
+   reading from ETS, message passing, deserialization and so on.
+
+(In the rest of the document we use just `dynamic()` as an abbreviation for
+`eqwalizer:dynamic()`.)
+
+The rationale for introducing `dynamic()` type is given in eqWAlizer
+[documentation](https://github.com/WhatsApp/eqwalizer/blob/main/docs/reference/modes.md#type-eqwalizerdynamic)
+and in the already mentioned resources for other languages and type checkers.
+
+We propose to extend Erlang surface syntax with new `dynamic()` builtin type
+to make Erlang more friendly to tooling built on ideas of gradual typing
+(like eqWAlizer) and ease its adoption
+(including using it for typing OTP libraries).
+
+Can it be just one type?
+========
+
+One may argue that just one type - `term()`/`any()` may be enough, and it's up
+to tooling to interpret it as a proper top type or as a special dynamic type
+for gradual typing. This, however, would be limiting in the long run.
+
```

Comment:
```suggestion

It has also [proven problematic in practice in the development of Gradualizer](https://github.com/josefs/Gradualizer/pull/244).
Originally, Gradualizer repurposed `any()` for the dynamic type and used `term()` as the top type.
It turned out to be confusing and incompatible with the preexisting use of these types.
A practical solution was to introduce a distinct `gradualizer:top()` type for the top of the subtyping lattice and treat `any()`/`term()` as the dynamic type.
```

Reply:
in practice > `, `in practice`,`
preexisting > `pre-existing`